onegov/ballot/collection.py

```diff
@@ -1,7 +1,75 @@
-from onegov.ballot.models import Ballot, Vote
+from onegov.ballot.models import Ballot, Election, Vote
 from sqlalchemy import cast, desc, extract, Integer
 
 
+class ElectionCollection(object):
+
+    def __init__(self, session, year=None):
+        self.session = session
+        self.year = year
+
+    def query(self):
+        return self.session.query(Election)
+
+    def for_year(self, year):
+        return self.__class__(self.session, year)
+
+    def get_latest(self):
+        """ Returns the elections with the latest date. """
+
+        latest_date = self.query().with_entities(Election.date)
+        latest_date = latest_date.order_by(desc(Election.date))
+        latest_date = latest_date.limit(1).first()
+
+        if not latest_date:
+            return None
+        else:
+            return self.by_date(latest_date)
+
+    def get_years(self):
+        """ Returns a list of years for which there are elections. """
+
+        query = self.query()
+        query = query.with_entities(
+            cast(extract('year', Election.date), Integer)
+        )
+        query = query.order_by(desc(Election.date))
+
+        return list(reversed(sorted((set(r[0] for r in query.all())))))
```

Comment:
I think you could use a [SELECT DISTINCT here instead](http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.distinct). And do it all on the database. That might be a bit faster.